Symfony/CS/Fixer/Contrib/UselessReturnFixer.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class UselessReturnFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $functionOpen = $tokens->getNextTokenOfKind($index, array('{'));
+            $this->fixFunction($tokens, $functionOpen, $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $functionOpen));
```

Comment:
1. iterate backward all tokens
2. for each T_FUNCTION iterate forward to find it's end
3. for each T_FUNCTION iterate backward from end to start

This means that for each function you are iterating through all its tokens... 3 times!
And or any 1st level nested function - 3x2, 2nd level nested function - 3x3 times and so on...

Here, you are not adding any tokens, only removing/merging. Then, you could Iterate forward on main loop and have no need to iterate few times over same tokens.